### PR TITLE
Add TicdcEventDataReader to parse ticdc data which consumed from kafka.

### PR DESCRIPTION
**Pull request by dixingxing0**
Add java demo to parse ticdc data which consumed from kafka.
This pr is just for review design, it's not the final version.

### CODE REVIEW

File: java-demo/src/main/java/org/pingcap/ticdc/cdc/TicdcEventFilter.java [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+    private Map<String, Long> tableMaxTSMap = new HashMap<>();
+    private Map<Integer, Long> resolveMaxTSMap = new HashMap<>();
+
+    public boolean filter(long rowChangeTS, int partition, String tableName) {
+        String mapKey = tableName + "_" + partition;
+        Long rowChangeMaxTS = tableMaxTSMap.get(mapKey);
+        if (rowChangeMaxTS == null) {
+            tableMaxTSMap.put(mapKey, rowChangeMaxTS);
+        } else {
+            if (rowChangeTS < rowChangeMaxTS) {
```
Comment (amyangfei) [inline java-demo/src/main/java/org/pingcap/ticdc/cdc/TicdcEventFilter.java:24]:
```suggestion
            if (rowChangeTS <= rowChangeMaxTS) {
```
Is it also a duplicated event If the `rowChangeTS` equals to `rowChangeMaxTS`

File: java-demo/src/main/java/org/pingcap/ticdc/cdc/TicdcEventFilter.java [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+                return false;
+            }
+            tableMaxTSMap.put(mapKey, rowChangeTS);
+        }
+
+        Long resolveMaxTS = resolveMaxTSMap.get(partition);
+        if (rowChangeMaxTS != null) {
+            if (rowChangeTS < resolveMaxTS) {
```
Comment (amyangfei) [inline java-demo/src/main/java/org/pingcap/ticdc/cdc/TicdcEventFilter.java:32]:
```suggestion
            if (rowChangeTS <= resolveMaxTS) {
```
The same as the comparison of `rowChangeMaxTS`

File: java-demo/src/main/java/org/pingcap/ticdc/cdc/KafkaMessage.java [hunk 3]
```diff
@@ -0,0 +1,57 @@
+package org.pingcap.ticdc.cdc;
```
Comment (liuzix) [inline java-demo/src/main/java/org/pingcap/ticdc/cdc/KafkaMessage.java:1]:
```suggestion
package com.pingcap.ticdc.cdc;
```
Probably should use our real domain name?

Comment (dixingxing0) [reply java-demo/src/main/java/org/pingcap/ticdc/cdc/KafkaMessage.java:1]: OK, I will fix it.

**Comment by dixingxing0** [comment]
> Besides, what about moving the demo dir to `demo/java`?
> Rest lgtm

**Pull request merged**